node/lib/util/push.js

```diff
@@ -236,7 +236,7 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
         const sha = pushMap[subName];
         const syntheticName =
                           SyntheticBranchUtil.getSyntheticBranchForCommit(sha);
-        const subRepo = yield SubmoduleUtil.getBareRepo(repo, subName);
+        const subRepo = yield SubmoduleUtil.getRepo(repo, subName);
```

Comment:
My understanding is that with getRepo, it get's the submodule repo with workdir() set (while using getBareRepo gets the repo in the meto repo's `.git/modules` which workdir() is undefined). Thus, the expectation here is that with getRepo, `subRepo` will have workdir() set and below git_util.push can then use the workdir().